src/bp/ui-shared/src/ContentForms/utils/typings.d.ts

```diff
@@ -0,0 +1,39 @@
+export interface MoreInfo {
+  label: string
+  url: string
+}
+
+export interface FormAdvancedSetting {
+  key: string
+  label: string
+  type: string
+  moreInfo?: MoreInfo
+}
+
+export interface Option {
+  value: string
+  label: string
+  related: FormField
+}
+
+export interface ContextMenu {
+  type: string
+  label: string
+}
+
+export interface FormField {
+  type: 'checkbox' | 'group' | 'select' | 'text' | 'textarea' | 'upload' | 'url'
+  key: string
+  label: string
+  placeholder?: string
+  addLabel?: string // when in a group, you have to specify the add button label
+  options?: Option[]
+  fields?: FormField[]
+  minimum?: number // when in a group, you can specify a minimum so the delete button won't show if there isn't more than the minimum
+  contextMenu?: ContextMenu[] // when in a group, you can add a contextual menu to add extra options
```

Comment:
If those 3 properties are only applicable to a group, then you could move them to a group object, ex:
```js
group: {
  addLabel:,
  // Hides the delete button when below that  
  minimum:
  contextMernu
}
```